src/GRA.Domain.Service/EmailService.cs

```diff
@@ -304,13 +310,22 @@ private async Task<EmailBase> GetEmailBase(int emailBaseId, int languageId)
 
             message.From.Add(new MailboxAddress(history.FromName, history.FromEmailAddress));
 
-            if (!string.IsNullOrEmpty(history.OverrideToEmailAddress))
+            try
             {
-                message.To.Add(MailboxAddress.Parse(history.OverrideToEmailAddress));
+                if (!string.IsNullOrEmpty(history.OverrideToEmailAddress))
+                {
+                    message.To.Add(MailboxAddress.Parse(history.OverrideToEmailAddress));
+                }
+                else
+                {
+                    message.To.Add(new MailboxAddress(history.ToName, history.ToEmailAddress));
+                }
             }
-            else
+            catch (ParseException ex)
             {
-                message.To.Add(new MailboxAddress(history.ToName, history.ToEmailAddress));
+                _logger.LogError("Unable to parse email address: {EmailAddress}",
+                    history.ToEmailAddress);
+                throw new GraException($"Unable to parse email address: {history.ToEmailAddress}", ex);
```

Comment:
This exception could bubble up to the method SendPickupEmailAsync in VendorCodeService which only has a try/catch at a very high level in the functions calling it.